### DISCUSSION TITLE
NetAlertX / Docker Desktop Windows 11 / Subnets

### DISCUSSION THREAD

**Discussion by JinxCH73** [netalertx/NetAlertX#861] [Q&A] [answered]
Hi !

Just manage to install NetAlertX on Docker Desktop (--net = host) for my W11.
I've read the subnets documentation but NetAlertX doesnt discover some hosts in my network...

When I check the System Info : 

services1 | 192.168.65.6/32 | 8,56 MB | 2,83 MB
lo | 127.0.0.1/8 | 34,42 MB | 34,42 MB
eth0 | 192.168.65.3/24 | 2,94 MB | 10,96 MB
docker0 | 172.17.0.1/16 | 0,00 MB | 0,00 MB

I dont see my LAN (192.168.0.0/24)... ??? And so I cant add the correct subnet of my WLAN connection.

Am I missing something ?

Thanks in advance.

**Comment by jokob-sk** [comment, marked as answer]
Hi @JinxCH73 ,

according to ChatGPT Win 11 doesn't support access to the network adapters: 

> Windows 11 does not natively support Docker's host network mode. This is because host mode allows containers to directly access the host machine's network stack, and Windows does not have the same network stack as Linux.
> 
> In Linux, Docker's host mode is commonly used because it allows the container to share the host's network stack, which can be beneficial for performance or specific networking configurations. However, on Windows, Docker runs on top of a virtual machine using Hyper-V or WSL2 (Windows Subsystem for Linux 2). Since the containers are running inside a VM, the concept of "host mode" networking becomes less practical, as the host in this case is the VM, not the Windows OS itself.


You can still theoretically use other scan methods that don't require network card access, such as DHCPLSS, PiHole, SNMP etc. Please check the plugins docs for alternatives: 

https://github.com/jokob-sk/NetAlertX/blob/main/front/plugins/README.md

Look at importer/scanner plugins 🔍/📥 - except for SNMPDEV and ARPSCAN as they require network access.

**Comment by JinxCH73** [comment]
Damn... OK, thanks !
